### PR TITLE
Reducer Modifier

### PR DESCRIPTION
**Pull request by markhalonen**
This is the safest/easiest modifier I could come up with. There is a new button in the viewer:

image

__Downsides__:
- It is a non-reversible filter. If you mess up, you have to remove the sample and add it again.
- This is not savable 
- Saving a Sample Session will save the wrong `beginTemp` and `endTemp`
   - I'm pretty sure this is currently wrong
- Do filter frequencies still make sense? I think it is still correct, it should adjust since the timestep changed.

We could also show the user more information about what their "Points To Keep" selection will do.. aka "This will result in a 10X speedup in calculations".

### CODE REVIEW

**Comment by bhalonen** [comment]
What units on frequency?
image

**Comment by markhalonen** [comment]
hertz. I can add the dim units label that is present in the sample creation page

File: Libraries/src/net/relinc/libraries/data/DataSubset.java [hunk 1]
```diff
@@ -19,6 +19,41 @@
...
+		
+		this.Data.timeData = newTimeData;
+		this.Data.data = newData;
+		
+		this.setBegin(this.getBegin() / reductionFactor);
+		this.setEnd(this.getEnd() / reductionFactor);
+		if(this.getBeginTemp() != null)
+			this.setBeginTemp(this.getBeginTemp() / reductionFactor);
+		if(this.getEndTemp() != null)
+			this.setEndTemp(this.getEnd() / reductionFactor);
```
Comment (markhalonen) [inline Libraries/src/net/relinc/libraries/data/DataSubset.java]: This is a bug, should be `this.getEndTemp()`. @bhalonen Should @glenhalonen fix?

Comment (bhalonen) [reply Libraries/src/net/relinc/libraries/data/DataSubset.java]: Yeah, @glenhalonen, see if you can resolve this bug.

**Pull request merged**